packages/app/src/app/overmind/internalActions.ts

```diff
@@ -508,14 +530,72 @@ export const showPrivacyPolicyNotification: Action = ({ effects, state }) => {
 const VIEW_MODE_DASHBOARD = 'VIEW_MODE_DASHBOARD';
 export const setViewModeForDashboard: Action = ({ effects, state }) => {
   const localStorageViewMode = effects.browser.storage.get(VIEW_MODE_DASHBOARD);
-  if (localStorageViewMode) {
+  if (localStorageViewMode === 'grid' || localStorageViewMode === 'list') {
     state.dashboard.viewMode = localStorageViewMode;
   }
 };
 
-export const setActiveTeamFromLocalStorage: Action = ({ effects, state }) => {
+export const setActiveTeamFromUrlOrStore: Action<void, string | null> = ({
+  actions,
+}) =>
+  actions.internal.setActiveTeamFromUrl() ||
+  actions.internal.setActiveTeamFromLocalStorage();
+
+export const setActiveTeamFromLocalStorage: Action<void, string | null> = ({
+  effects,
+  actions,
+}) => {
   const localStorageTeam = effects.browser.storage.get(TEAM_ID_LOCAL_STORAGE);
-  if (localStorageTeam) {
-    state.activeTeam = localStorageTeam;
+
+  if (typeof localStorageTeam === 'string') {
+    actions.setActiveTeam({ id: localStorageTeam });
+    return localStorageTeam;
   }
+
+  return null;
+};
+
+export const setActiveTeamFromUrl: Action<void, string | null> = ({
+  effects,
+  actions,
+}) => {
+  const currentUrl =
+    typeof document === 'undefined' ? null : document.location.href;
```

Comment:
```suggestion
  const currentUrl = document?.location.href;
```

Reply:
Referring to `document` on the server throws an error, that's why I started to use `typeof` for checks on globals.

![image](https://user-images.githubusercontent.com/587016/86466501-06e76b00-bd34-11ea-9933-bb64e84a74c6.png)